exercises/prime-factors/src/example.h

```diff
@@ -0,0 +1,11 @@
+#ifndef PRIME_FACTORS_H
+#define PRIME_FACTORS_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#define MAXFACTORS 10
+
+size_t find_factors(uint64_t n, uint64_t * factors);
```

Comment:
I believe you can use `uint64_t factors[static MAXFACTORS]` to require that `factors` have at least 10 elements.

Reply:
This is also new to me. For the sake of understanding - what could have happen if the size was declared without ```static```?

Reply:
AS the comments on that blog post and the second paragraph [here][1] remark A) this is not supported by many compilers (e.g. GCC doesn't), B) if the guarantee made by the programmer is not honored by the caller then the behaviour is undefined, and C) the optimization gains are only apparent on some architectures

On the other hand, using the `const` qualifier _does_ appear to work on GCC (>=v7.2.0 at least).

[1]: https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_71/rzarg/static_array_index.htm

Reply:
@wolf99 thanks for the info. I've never actually used this before (just picked it up from someone well-versed in C standards trivia on Stack Overflow) so wasn't sure about the exact behavior. Sounds like this isn't terribly useful in practice. Thanks for doing some digging.